provider/tencentcloud/dnspod.go

```diff
@@ -0,0 +1,263 @@
+			}
+		}
+		totalCount = *response.Response.RecordCountInfo.TotalCount
+		request.Offset = common.Uint64Ptr(*request.Offset + uint64(len(response.Response.RecordList)))
+	}
+	return domainList, nil
+}
+
+type RecordListGroup struct {
+	Domain     *dnspod.DomainListItem
+	RecordList []*dnspod.RecordListItem
+}
+
+func (p *TencentCloudProvider) applyChangesForDNS(changes *plan.Changes) error {
+	recordsGroupMap, err := p.recordsForDNS()
+	if err != nil {
+		return err
+	}
+
+	zoneNameIDMapper := provider.ZoneIDName{}
+	for _, recordsGroup := range recordsGroupMap {
+		if recordsGroup.Domain.DomainId != nil {
+			zoneNameIDMapper.Add(strconv.FormatUint(*recordsGroup.Domain.DomainId, 10), *recordsGroup.Domain.Name)
+		}
+	}
+
+	// Apply Change Delete
+	deleteEndpoints := make(map[string][]uint64)
+	for _, change := range [][]*endpoint.Endpoint{changes.Delete, changes.UpdateOld} {
+		for _, deleteChange := range change {
+			if zoneId, _ := zoneNameIDMapper.FindZone(deleteChange.DNSName); zoneId != "" {
+				zoneIdString, _ := strconv.ParseUint(zoneId, 10, 64)
+				recordListGroup := recordsGroupMap[zoneIdString]
+				for _, domainRecord := range recordListGroup.RecordList {
+					subDomain := getSubDomain(*recordListGroup.Domain.Name, deleteChange)
+					if *domainRecord.Name == subDomain && *domainRecord.Type == deleteChange.RecordType {
+						for _, target := range deleteChange.Targets {
+							if *domainRecord.Value == target {
+								if _, exist := deleteEndpoints[*recordListGroup.Domain.Name]; !exist {
+									deleteEndpoints[*recordListGroup.Domain.Name] = make([]uint64, 0)
+								}
+								deleteEndpoints[*recordListGroup.Domain.Name] = append(deleteEndpoints[*recordListGroup.Domain.Name], *domainRecord.RecordId)
+							}
+						}
+					}
+				}
+			}
+		}
+	}
+
+	if err := p.deleteRecords(deleteEndpoints); err != nil {
+		return err
+	}
+
+	// Apply Change Create
+	createEndpoints := make(map[string][]*endpoint.Endpoint)
+	for zoneId, _ := range zoneNameIDMapper {
+		createEndpoints[zoneId] = make([]*endpoint.Endpoint, 0)
+	}
+	for _, change := range [][]*endpoint.Endpoint{changes.Create, changes.UpdateNew} {
+		for _, createChange := range change {
+			if zoneId, _ := zoneNameIDMapper.FindZone(createChange.DNSName); zoneId != "" {
+				createEndpoints[zoneId] = append(createEndpoints[zoneId], createChange)
+			}
+		}
+	}
+	if err := p.createRecord(recordsGroupMap, createEndpoints); err != nil {
+		return err
+	}
+	return nil
+}
+
+func (p *TencentCloudProvider) createRecord(zoneMap map[uint64]*RecordListGroup, endpointsMap map[string][]*endpoint.Endpoint) error {
+	for zoneId, endpoints := range endpointsMap {
+		zoneIdString, _ := strconv.ParseUint(zoneId, 10, 64)
+		domain := zoneMap[zoneIdString]
+		for _, endpoint := range endpoints {
+			for _, target := range endpoint.Targets {
+				if endpoint.RecordType == "TXT" && strings.HasPrefix(target, "\"heritage=") {
+					target = strings.Trim(target, "\"")
```

Comment:
I am just scrolling the code a bit and this line and the above looks a bit weird to me